### PR TITLE
add incremental node build when we append some modifications (copy or cut and paste)

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 1]
```diff
@@ -1332,9 +1337,11 @@ public String duplicateModifications(UUID studyUuid, UUID nodeUuid, List<UUID> m
         String response;
         try {
             checkStudyContainsNode(studyUuid, nodeUuid);
-            UUID targetGroupUuid = networkModificationTreeService.getModificationGroupUuid(nodeUuid);
-            response = networkModificationService.duplicateModification(targetGroupUuid, modificationUuidList);
-            updateStatuses(studyUuid, nodeUuid, false);
+            NodeModificationInfos nodeInfos = networkModificationTreeService.getNodeModificationInfos(nodeUuid);
+            UUID networkUuid = networkStoreService.getNetworkUuid(studyUuid);
+            response = networkModificationService.duplicateModification(modificationUuidList, networkUuid, nodeInfos);
+            // no invalidation cause we will apply the duplicated modifications on a built node
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:1343]: Copy/Paste of a modification inside the same node should invalidate the children nodes of this node

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Tests: NOK

File: src/main/java/org/gridsuite/study/server/service/StudyService.java (hunk 1 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:1343]: yes, I missed this point!

Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:1343]: this is also true for some cut/paste (when we have to build, between 2 unrelated subtrees)

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 2]
```diff
@@ -1803,11 +1808,16 @@ public void testReorderModification() throws Exception {
...
+        wireMock.verify(1, WireMock.putRequestedFor(WireMock.urlMatching(
+                        "/v1/groups/" + modificationNode.getModificationGroupUuid() + "\\?action=MOVE&networkUuid=" + NETWORK_UUID_STRING +
+                                "&reportUuid=(.*)\\" +
+                                "&reporterId=" + modificationNode.getId() +
+                                "&variantId=" + VARIANT_ID +
+                                "&originGroupUuid=" + modificationNode.getModificationGroupUuid() + "&build=false&before=" + modification2 + "$"))
                 .withRequestBody(WireMock.equalToJson(expectedBody))); // modification1 is still in the request body
     }
 
+    /* TODO
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:1820]: yes, I have to fix these 2 tests.

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 3]
```diff
@@ -1066,7 +1070,8 @@ public ResponseEntity<Void> buildNode(@Parameter(description = "Study uuid") @Pa
                            @ApiResponse(responseCode = "404", description = "The study or node doesn't exist")})
     public ResponseEntity<Void> stopBuild(@Parameter(description = "Study uuid") @PathVariable("studyUuid") UUID studyUuid,
                                                       @Parameter(description = "nodeUuid") @PathVariable("nodeUuid") UUID nodeUuid) {
-        studyService.stopBuild(studyUuid, nodeUuid);
+        // TODO studyUuid parameter is not used
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/StudyController.java:1073]: I think it is a real TODO, and the REST api should be changed

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 4]
```diff
@@ -1332,9 +1343,11 @@ public String duplicateModifications(UUID studyUuid, UUID nodeUuid, List<UUID> m
         String response;
         try {
             checkStudyContainsNode(studyUuid, nodeUuid);
-            UUID targetGroupUuid = networkModificationTreeService.getModificationGroupUuid(nodeUuid);
-            response = networkModificationService.duplicateModification(targetGroupUuid, modificationUuidList);
-            updateStatuses(studyUuid, nodeUuid, false);
+            NodeModificationInfos nodeInfos = networkModificationTreeService.getNodeModificationInfos(nodeUuid);
+            UUID networkUuid = networkStoreService.getNetworkUuid(studyUuid);
+            response = networkModificationService.duplicateModification(modificationUuidList, networkUuid, nodeInfos);
+            // invalidate the whole subtree except the target node (could be incrementally built)
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:1349]: ok I will clarify

**Review by FranckLecuyer — APPROVED** [review]

**Review by archendel — APPROVED** [review]

**Pull request merged**